make_site.py

```diff
@@ -218,6 +226,13 @@ class Course:
 
 if DOWNLOAD:
     print('Downloading header-data.json...') #  This is a slow operation, so let's inform readers.
+    # header-data.json contains information for every single declaration in mathlib
+    # which has a generated documentation entry (e.g., skipping private declarations by default).
+    # The resulting file is huge (several hundred MB), hence we use a small HACK:
+    # doc-gen4 uploads this file as a .bmp file, so browsers will download it compressed,
+    # and employ their machinery for dealing with large files. XXX is this true at all?
+    # When downloading it locally, we save it with the correct extension.
+    # why? blocklisting of file extensions? automatic compression in web browsers?
```

Comment:
```suggestion
    # doc-gen4 uploads this file as a .bmp file, so GitHub's servers will serve it in
    # compressed form. When downloading it locally, we save it with the correct extension, 
    # so that we can more easily inspect the file locally for debugging.
```
Here are my best guesses to the answers to your questions.

Reply:
That looks very plausible - reading the linked discussion, that seems to be it. I've tweaked this slightly and adopted it. Thanks!